src/expressions/dataTypes/casting/tryCastToType.ts

```diff
@@ -26,18 +26,20 @@ import castToGYear from './castToGYear';
 import castToGYearMonth from './castToGYearMonth';
 import castToHexBinary from './castToHexBinary';
 import castToInteger from './castToInteger';
+import castToNumeric from './castToNumeric';
 import castToString from './castToString';
 import castToTime from './castToTime';
 import castToUntypedAtomic from './castToUntypedAtomic';
 import castToYearMonthDuration from './castToYearMonthDuration';
 
 const TREAT_AS_PRIMITIVE = [
+	ValueType.XSNUMERIC,
```

Comment:
Are we sure this is a primitive or was this added as a temporary fix?

Reply:
Numeric is a union type, so this is a temporary fix. I'll make it an issue

Reply:
Perfect!